great_expectations/core/serializer.py

```diff
@@ -0,0 +1,62 @@
+"""Serializer class interface definition.
+
+Serializers determine how to write an object to disk, json, etc.
+A serializer comprises the object destination and name e.g. YAMLReadyDictDatasourceConfigSerializer.
+A base implementation (DictConfigSerializer) is provided if no modification needs to be included for the specific object / destination pair.
+
+Typical usage example:
+
+datasource_config = DatasourceConfig(...)
+serializer = DictConfigSerializer(schema=datasourceConfigSchema)
+serialized_value = serializer.serialize(datasource_config)
+"""
+
+import abc
+from typing import Optional, Union
+
+from great_expectations.core.configuration import AbstractConfig
+from great_expectations.marshmallow__shade import Schema
+
+
+class AbstractConfigSerializer(abc.ABC):
+    """Serializer interface.
+
+    Note: When mypy coverage is enhanced further, this Abstract class can be replaced with a Protocol.
+    """
+
+    def __init__(self, schema: Optional[Schema] = None) -> None:
```

Comment:
Is this `schema` ever actually optional?

Reply:
I agree - seems like it is necessary

Reply:
No, it's just provided / overridden under the hood by some subclasses. Is there a better pattern for that? Like a builder class or method? Or should we just always pass it in.

Reply:
I was kind of thinking we could decouple our usage of marshmallow with these serializers, in case we want to use a different library. But we can make it optional in the future if necessary, harder to go the other way.

Reply:
Removed Optional and added to all constructor calls. Leaving this comment unresolved in case there is further discussion of above replies.